### PR TITLE
[AirQuality] Bugfix on issues #3281, #3114, #2422, #2141

### PR DESCRIPTION
**Pull request by clinique**
Also adds :
 - usage of QuantityType for Pressure, Temperature and Humidity
 - Autodiscovery
 - attempt to workaround issue #3488

Documentation update

### CODE REVIEW

File: addons/binding/org.openhab.binding.airquality/ESH-INF/thing/thing-types.xml [hunk 1]
```diff
@@ -56,112 +57,117 @@
...
-		<state readOnly="true" pattern="%s"></state>
+		<state readOnly="true" pattern="%s"/>
 	</channel-type>
 
 	<channel-type id="pm25">
 		<item-type>Number</item-type>
 		<label>PM2.5</label>
 		<description>Fine particles pollution level</description>
 		<category>PM2.5</category>
-		<state readOnly="true" pattern="%d μg/m&#179;"></state>
```
Comment (martinvw) [inline addons/binding/org.openhab.binding.airquality/ESH-INF/thing/thing-types.xml]: Why does the unit disappear here?

Comment (martinvw) [reply addons/binding/org.openhab.binding.airquality/ESH-INF/thing/thing-types.xml]: Applies to the next items as well

File: addons/binding/org.openhab.binding.airquality/ESH-INF/thing/thing-types.xml [hunk 2]
```diff
@@ -56,112 +57,117 @@
...
-		<label>Humidity level</label>
-		<description></description>
-		<category>Humidity level</category>
-		<state readOnly="true" min="0" max="100" pattern="%.2f %%">
-		</state>
+		<item-type>Number:Dimensionless</item-type>
+		<label>Humidity</label>
+		<description>Current humidity</description>
+		<category>Humidity</category>
+		<state readOnly="true" min="0" max="100" pattern="%.2f %unit%" />
```
Comment (martinvw) [inline addons/binding/org.openhab.binding.airquality/ESH-INF/thing/thing-types.xml]: Does the combination of Dimensionless and unit work?

File: addons/binding/org.openhab.binding.airquality/src/main/java/org/openhab/binding/airquality/internal/discovery/AirQualityDiscoveryService.java [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+                    previousLocation = currentLocation;
+                }
+            }, 0, LOCATION_CHANGED_CHECK_INTERVAL, TimeUnit.SECONDS);
+            logger.debug("Scheduled Air Qualitylocation-changed job every {} seconds", LOCATION_CHANGED_CHECK_INTERVAL);
+        }
+    }
+
+    public void createResults(PointType location) {
+        ThingUID localAirQualityThing = new ThingUID(THING_TYPE_AQI, LOCAL);
+        Map<String, Object> properties = new HashMap<>(3);
```
Comment (martinvw) [inline addons/binding/org.openhab.binding.airquality/src/main/java/org/openhab/binding/airquality/internal/discovery/AirQualityDiscoveryService.java]: Can you leave out the 3 here, it is some strange optimization now for a map which contains one item.

File: addons/binding/org.openhab.binding.airquality/src/main/java/org/openhab/binding/airquality/internal/discovery/AirQualityDiscoveryService.java [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+
+    @Override
+    protected void startBackgroundDiscovery() {
+        if (discoveryJob == null) {
+            discoveryJob = scheduler.scheduleWithFixedDelay(() -> {
+                PointType currentLocation = locationProvider.getLocation();
+                if (!Objects.equals(currentLocation, previousLocation)) {
+                    logger.debug("Location has been changed from {} to {}: Creating new discovery results",
+                            previousLocation, currentLocation);
+                    createResults(currentLocation);
```
Comment (martinvw) [inline addons/binding/org.openhab.binding.airquality/src/main/java/org/openhab/binding/airquality/internal/discovery/AirQualityDiscoveryService.java]: Should it clear the results from the previous location?

File: addons/binding/org.openhab.binding.airquality/src/main/java/org/openhab/binding/airquality/internal/discovery/AirQualityDiscoveryService.java (hunk 3 above)
Comment (clinique) [reply addons/binding/org.openhab.binding.airquality/src/main/java/org/openhab/binding/airquality/internal/discovery/AirQualityDiscoveryService.java]: Yes, well spotted.

File: addons/binding/org.openhab.binding.airquality/src/main/java/org/openhab/binding/airquality/internal/discovery/AirQualityDiscoveryService.java (hunk 4 above)
Comment (clinique) [reply addons/binding/org.openhab.binding.airquality/src/main/java/org/openhab/binding/airquality/internal/discovery/AirQualityDiscoveryService.java]: I would not like to delete previous location automatically. I think it's better to leave it to the user - or I do not get your point.

**Comment by clinique** [comment]
@martinvw : regarding the retry. It seems that the API returns empty result from time to time, this puts the binding in offline mode. My attempt is just to retry at least once before going offline. Not easy to reproduce nor to test :(

File: addons/binding/org.openhab.binding.airquality/src/main/java/org/openhab/binding/airquality/internal/discovery/AirQualityDiscoveryService.java (hunk 4 above)
Comment (martinvw) [reply addons/binding/org.openhab.binding.airquality/src/main/java/org/openhab/binding/airquality/internal/discovery/AirQualityDiscoveryService.java]: No I meant to delete the unapproved discovery result, but its not blocking for me.

File: addons/binding/org.openhab.binding.airquality/ESH-INF/thing/thing-types.xml (hunk 1 above)
Comment (martinvw) [reply addons/binding/org.openhab.binding.airquality/ESH-INF/thing/thing-types.xml]: So its actually the core of the issue, thanks!

**Comment by martinvw** [comment]
Thanks!

**Pull request merged**